test/Archive/ArchiveFile.test.cpp

```diff
@@ -0,0 +1,69 @@
+#include "Archive/VolFile.h"
+#include "Archive/ClmFile.h"
+#include "XFile.h"
+#include <gtest/gtest.h>
+#include <vector>
+#include <string>
+
+void TestEmptyArchive(Archive::ArchiveFile& archiveFile, const std::string& archiveFilename);
+
+
+TEST(VolFile, EmptyArchive) 
+{
+	const std::string archiveFilename("EmptyArchive.vol");
+
+	{
+		// Create an empty archive
+		const std::vector<std::string> filesToPack;
+		Archive::VolFile::CreateArchive(archiveFilename, filesToPack);
+
+		Archive::VolFile archiveFile(archiveFilename);
+
+		SCOPED_TRACE("Empty Volume File");
+		TestEmptyArchive(archiveFile, archiveFilename);
+
+		EXPECT_THROW(archiveFile.GetCompressionCode(0), std::runtime_error);
+	}
+
+	XFile::DeletePath(archiveFilename);
+}
+
+TEST(ClmFile, EmptyArchive) 
+{
+	const std::string archiveFilename("EmptyArchive.clm");
+
+	{
+		// Create an empty archive
+		const std::vector<std::string> filesToPack;
+		Archive::ClmFile::CreateArchive(archiveFilename, filesToPack);
+
+		Archive::ClmFile archiveFile(archiveFilename);
+
+		SCOPED_TRACE("Empty CLM File");
+		TestEmptyArchive(archiveFile, archiveFilename);
+	}
+
+	XFile::DeletePath(archiveFilename);
+}
+
+// Deletes any files extracted during the test
+void TestEmptyArchive(Archive::ArchiveFile& archiveFile, const std::string& archiveFilename)
+{
+	const std::string extractDirectory("./TestExtract");
+
+	EXPECT_THROW(archiveFile.GetIndex("TestItem"), std::runtime_error);
+	EXPECT_THROW(archiveFile.GetName(0), std::runtime_error);
+	EXPECT_THROW(archiveFile.GetSize(0), std::runtime_error);
+	EXPECT_FALSE(archiveFile.Contains("TestItem"));
+
+	EXPECT_THROW(archiveFile.OpenStream(0), std::runtime_error);
+	EXPECT_THROW(archiveFile.ExtractFile(0, "Test"), std::runtime_error);
```

Comment:
The above checks seem a little bit out of scope. Again, this is largely just checking `EXPECT_EQ(0, archiveFile.GetCount());`, which is already checked elsewhere. Granted, tests like these are important, to ensure there are no bounds overflow in relation to `GetCount`, I'm just not certain this is the best place for it. I sort of feel these should be split off to a separate unit test, which is more targeted towards ensuring access past `GetCount` results in an exception. That may include a case where `GetCount` is 0, or other values.

Reply:
Agreed more tests should be made against volume and clm files which may result in pulling some of the tests out of the empty archive test routines. I wasn't planning to write robust tests against everything clm/vol related right now, although it would be a good goal.